### PR TITLE
Code generator for haskell-servant framework

### PR DESCRIPTION
**Pull request by algas**

### CODE REVIEW

**Comment by wing328** [comment]
@algas thanks for the PR! :beers: 

One suggestion is to rename `haskell` to `haskell-servant` instead as developers may confuse `haskell` with the Haskell API client. (we'd the issue with `nodejs` in which developers expected a Nodejs API client but `nodejs` refers to Nodejs server generator instead)

**Comment by algas** [comment]
@wing328 Thank you for your advice. I renamed it.

File: bin/haskell-petstore.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+executable="./modules/swagger-codegen-cli/target/swagger-codegen-cli.jar"
+
+if [ ! -f "$executable" ]
+then
+  mvn clean package
+fi
+
+# if you've executed sbt assembly previously it will use that instead.
+export JAVA_OPTS="${JAVA_OPTS} -XX:MaxPermSize=256M -Xmx1024M -DloggerPath=conf/log4j.properties"
+ags="$@ generate -t modules/swagger-codegen/src/main/resources/haskell -i modules/swagger-codegen/src/test/resources/2_0/petstore.json -l haskell -o samples/server/petstore/haskell"
```
Comment (wing328) [inline bin/haskell-petstore.sh]: @algas  should `-l haskell` be renamed to `-l hashkell-servant` and `samples/server/petstore/haskell` be renamed to `samples/server/petstore/haskell-servant`?

**Comment by wing328** [comment]
PR merged. Thanks again for the contribution.

**Pull request merged**